File: shared/src/commonMain/kotlin/com/ojoclaro/shared/model/AssistantModels.kt
package com.ojoclaro.shared.model

import kotlinx.serialization.Serializable

@Serializable
data class AssistRequest(
    val command: AppCommand,
    val userMessage: String,
    val imageBase64: String? = null,
    val location: GeoPoint? = null,
    val deviceLocale: String = "es-AR",
    val accessibilityMode: AccessibilityMode = AccessibilityMode.VOICE_FIRST
)

@Serializable
data class GeoPoint(
    val latitude: Double,
    val longitude: Double,
    val accuracyMeters: Double? = null
)

@Serializable
enum class AccessibilityMode {
    VOICE_FIRST,
    LOW_VISION_HIGH_CONTRAST,
    FAMILY_SETUP
}

@Serializable
data class AssistResponse(
    val spokenText: String,
    val shortText: String,
    val confidence: ConfidenceLevel,
    val category: ResponseCategory,
    val safetyNotice: String? = null,
    val suggestedActions: List<SuggestedAction> = emptyList()
)

@Serializable
enum class ConfidenceLevel {
    HIGH,
    MEDIUM,
    LOW
}

@Serializable
enum class ResponseCategory {
    TEXT_READING,
    SCENE_DESCRIPTION,
    PRODUCT_IDENTIFICATION,
    DOCUMENT_SUMMARY,
    LOCATION_HELP,
    EMERGENCY,
    SYSTEM
}

@Serializable
data class SuggestedAction(
    val id: String,
    val label: String,
    val commandType: AppCommandType
)

File: ios/OjoClaroIOS/Models/AssistantModels.swift
import Foundation

enum AppCommandType: String, Codable {
    case describeScene = "DESCRIBE_SCENE"
    case readText = "READ_TEXT"
    case readDocument = "READ_DOCUMENT"
    case identifyProduct = "IDENTIFY_PRODUCT"
    case emergencyHelp = "EMERGENCY_HELP"
    case currentLocation = "CURRENT_LOCATION"
    case unknown = "UNKNOWN"
}

struct AppCommand: Codable {
    let type: AppCommandType
    let originalText: String
    let locale: String
    let requiresCamera: Bool
    let requiresLocation: Bool
    let highRisk: Bool
}

struct AssistRequest: Codable {
    let command: AppCommand
    let userMessage: String
    let imageBase64: String?
    let location: GeoPoint?
    let deviceLocale: String
    let accessibilityMode: String
}

struct GeoPoint: Codable {
    let latitude: Double
    let longitude: Double
    let accuracyMeters: Double?
}

struct AssistResponse: Codable {
    let spokenText: String
    let shortText: String
    let confidence: String
    let category: String
    let safetyNotice: String?
    let suggestedActions: [SuggestedAction]
}

struct SuggestedAction: Codable {
    let id: String
    let label: String
    let commandType: AppCommandType
}
